### PR TITLE
Command check file save before open

### PR DESCRIPTION
**Pull request by mjkkirschner**
### Purpose

http://adsk-oss.myjetbrains.com/youtrack/issue/MAGN-10495

1 - This PR adds another command to the `DynamoViewModel` - `OpenIfSaved` command which wraps a `DynamoModel` `OpenFileCommand` with a check for unsaved changes on the home workspace. If there are unsaved changes the message box requesting a save is popped up.
__________
2 - This task also requested exploring a property exposing the modal state of the Dynamo UI - I.E, are there any windows open that would disallow interactions with the `DynamoModel`.

the best I have found is the static class `ComponentDispatcher.EnterThreadModal` event which would allow us to set a bool flag somewhere on the view or view model. Unfortunately message boxes(without owners) do not fire this event so a scan over all message box and other popup / window code would need to be done and consequences of this would need to be tested.
```
 System.Windows.Interop.ComponentDispatcher.EnterThreadModal += (o,e) {modal = true;}

```
I would guess this handler would be set at view model or view construction time. Looking for thoughts and feedback on this, note this PR does not include this handler.

**For now, I will leave this property as a followup task if deemed necessary**

### Declarations

Check these if you believe they are true

- [x] The code base is in a better state after this PR
- [x] Is documented according to the [standards](https://github.com/DynamoDS/Dynamo/wiki/Coding-Standards)
- [x] The level of testing this PR includes is appropriate
- [ ] User facing strings, if any, are extracted into `*.resx` files
- [ ] All tests pass using the self-service CI.
- [ ] Snapshot of UI changes, if any.

### Reviewers

@ramramps 
@sharadkjaiswal 
### FYIs

@kronz @BogdanZavu

### CODE REVIEW

File: src/DynamoCoreWpf/ViewModels/Core/DynamoViewModelDelegateCommands.cs [hunk 1]
```diff
@@ -80,7 +82,7 @@ private void InitializeDelegateCommands()
             ShowNewPresetsDialogCommand = new DelegateCommand(ShowNewPresetStateDialogAndMakePreset, CanShowNewPresetStateDialog);
             NodeFromSelectionCommand = new DelegateCommand(CreateNodeFromSelection, CanCreateNodeFromSelection);            
        }
-
+        public DelegateCommand OpenIfSavedCommand { get; set; }
```
Comment (ramramps) [inline src/DynamoCoreWpf/ViewModels/Core/DynamoViewModelDelegateCommands.cs]: where is this attached to XAML?

Comment (mjkkirschner) [inline src/DynamoCoreWpf/ViewModels/Core/DynamoViewModelDelegateCommands.cs]: nope

File: src/DynamoCoreWpf/ViewModels/Core/DynamoViewModel.cs [hunk 2]
```diff
@@ -1138,6 +1138,34 @@ public FileDialog GetSaveDialog(WorkspaceModel workspace)
         }
 
         /// <summary>
+        /// Attempts to open a file using the passed open command, but wraps the call
+        /// with a check to make sure no unsaved changes to the HomeWorkspace are lost.
+        /// </summary>
+        /// <param name="openCommand"> <see cref="DynamoModel.OpenFileCommand"/> </param>
+        private void OpenIfSaved(object openCommand)
+        {
```
Comment (ramramps) [inline src/DynamoCoreWpf/ViewModels/Core/DynamoViewModel.cs]: Is this command specific to Revit playlist ? If OpenFileCommand is a delegate, can't we just do a check there?
may be,  I am dumb asking this question

Comment (mjkkirschner) [inline src/DynamoCoreWpf/ViewModels/Core/DynamoViewModel.cs]: I don't want to change the behavior of the openFileCommand, I tested this first and it alters behavior in a bunch of places... throws multiple dialogs, etc etc.

Comment (ramramps) [inline src/DynamoCoreWpf/ViewModels/Core/DynamoViewModel.cs]: Ok. LGTM then

**Comment by ramramps** [comment]
LGTM.

**Comment by mjkkirschner** [comment]
I'm going to merge this in - @BogdanZavu please test this API and let us know if it does what you need.

**Comment by BogdanZavu** [comment]
Thank you @mjkkirschner , @ramramps . I'll test it tomorrow and let you know.

**Pull request merged**